server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -534,328 +559,331 @@ describe("Routerlicious", () => {
+											);
+											checkNack(clients[0], "Invalid signal message");
+										});
+									});
+								} else if (description === "with v1 and v2 clients") {
+									it("should broadcast signal from a v1 client to all connected clients", () => {
+										const expectedSignal = sendValidAndReturnExpectedSignal(
+											clients[0],
+											stringSignalContent,
+										);
+
+										verifyExpectedClientSignals(clients, [expectedSignal]);
+									});
+
+									it("should broadcast signal from a v2 client to all connected clients", () => {
+										const targetedSignal: ISentSignalMessage = {
+											content: stringSignalContent,
+										};
+
+										const expectedSignal = sendAndReturnExpectedSignal(
+											clients[1],
+											targetedSignal,
+										);
+
+										verifyExpectedClientSignals(clients, [expectedSignal]);
+									});
+
+									it("can target a v1 client from a v2 client", () => {
+										const targetedSignal: ISentSignalMessage = {
+											targetClientId: clients[0].clientId,
+											content: stringSignalContent,
+										};
+
+										const expectedSignal = sendAndReturnExpectedSignal(
+											clients[1],
+											targetedSignal,
+										);
+
+										verifyExpectedClientSignals([clients[0]], [expectedSignal]);
+										verifyExpectedClientSignals(
+											clients.filter((client) => client !== clients[0]),
+											[],
+										);
+									});
+								}
 							});
-						});
-						describe("sending two signals", () => {
-							it("can transmit both targeted and broadcast signals", () => {
-								const targetedSignal: ISentSignalMessage = {
-									targetClientId: clients[0].clientId,
-									content: "TargetedSignal",
-								};
-								const broadcastSignal: ISentSignalMessage = {
-									content: "BroadcastSignal",
-								};
-
-								const expectedTargetedSignal = sendAndReturnExpectedSignal(
-									clients[1],
-									targetedSignal,
-								);
-								const expectedBroadcastSignal = sendAndReturnExpectedSignal(
-									clients[1],
-									broadcastSignal,
-								);
+							describe("sending multiple signals", () => {
+								[0, 1].forEach((clientIndex) => {
+									const fromClient = `from client ${clientIndex} (v${clientVersion[clientIndex]})`;
+									it(`${fromClient} should broadcast signals sent from multiple clients to all connected clients`, () => {
+										const firstSignal = sendValidAndReturnExpectedSignal(
+											clients[clientIndex],
+											"first signal",
+										);
+										const secondSignal = sendValidAndReturnExpectedSignal(
+											clients[clientIndex ^ 1],
+											"second signal",
+										);
+
+										verifyExpectedClientSignals(clients, [
+											firstSignal,
+											secondSignal,
+										]);
+									});
+									it(`${fromClient} should broadcast multiple batched signals sent from single clients to all connected clients`, () => {
```

Comment:
This doesn't look like batched signals. It looks like two signals. If it were batched, then you could add a check in between each of the `sendValid*` calls that no signals have been sent back.

Reply:
Oh I see. From my understanding signal batching is not a feature that afr/r11s provides or supports. They mention this is a feature they plan to adopt in the future

Reply:
Added batching support to both v1, v2 and v1v2 mix